FeatuePanel: Redesign the share dialog

Dlurak opened the pull request:
### Description

This fixes #692 as that issue is quite long and has a lot of comments it is possible that I forgot about something.

#### New chips

Under the heading we now have a few chips for common actions.

![image](https://github.com/user-attachments/assets/dda140ea-c8b2-4b79-9e93-10eca3e81239)
![image](https://github.com/user-attachments/assets/a4d57980-13b2-4580-a3a9-a6fe098546ed)

#### Share dialog

The share dialog offers some primary options  for links to other services or to share the shortend osmapp link options

Under that there is a tab view to either open another service on the coordinates/feature, this list is exactly like the old coordinate menu or the option to share these urls or the coordinates  using the native share menu, if the webshare api isn't available we fallback to copying which is shown by the tabs label.

Under that there is a collapsible container with attributions to the images at the top.

![image](https://github.com/user-attachments/assets/c70169d1-9418-4929-8c28-f742242a4e65)

Dlurak commented:
@govvin and @jvaclavik what do you think?

jvaclavik commented:
I like it very much!
I have a few thoughts:
- buttons have no on touch state and sometimes it selected text of the button, but both can be fixed by using mui chips
- there is a bug when quick buttons doesn't fit properly and you can't scroll. It's cut 
![image](https://github.com/user-attachments/assets/a66078a4-80f4-48b3-ba70-371b879ed016)

- i was really confused by osmapp in icon buttons. It's just share not a link. Below i suggest solution
- i think apple maps are too prominent, suggesting move them next to google maps 👮‍♀️😂
- icons for external link and share in icon button in sharing dialog should be after the text. It will look better. In list it's correct
- do we need image attribution? And if yes, i suggest to make it smaller, without bold, with secondary color and some margin top. User doesn't need it and it's very prominent. - i tested it only on mobile, not sure what's the behaviour of share on desktop
- one idea to consider: maybe tabs should include also icon buttons. In this case we could show different content so osmapp would fit there. In this case i suggest to show share tab first and osmapp should have the first position
- Also I'm not sure if we need all the duplicity links there between the list and icon buttons. But we can update it later. 
- What about rename sections to:
Link -> Open in
Share -> Share a link (this should be default and the first tab IMHO)


Good job @Dlurak 🥳! I really like changes you are making in your PRs.

Dlurak commented:
- I've migrated to mui chips
- Fixed the scrolling bug
- I moved apple maps behind google maps
- The icon is now behind the label
- I'm not sure but I think we need the attribution, it was indeed very prominent I made it smaller and in a secondary color
- I've renamed the tabs, added German translations and included icons
- The default is now sharing except when the webshare api isn't available then the links are the default

---

I'm not sure if moving the icon buttons into the tab views would be a really good idea.

govvin commented:
Awesome job @Dlurak !  :tada:  Thank you.

Some thoughts:
- Tabview labels
  - User-friendly labels are desirable. "GeoURI" is not a familiar term for most people. How about: "Native App",  "Open in App" or "Open in Device",
    - Maybe it needs a better icon, for example, something similar to [this](https://thenounproject.com/icon/mapping-4733828/) ?
  - the icons look too big (IMO), and if we choose the right icon, the labels below them _should_ be unnecessary. Only first 3 icons are visible, and the rest is accessible by scrolling first (is that a behavior we want?, when same option is already listed  below.
  - Smaller icons, without labels, can fit 4-5 items without the need to scroll.
- Attribution
  - The attribution doesn't have to be in that window. You can follow attribution requirements by including it in the main attribution (lower right corner of the main map.)  In addition, the "i" button also mention where the specific image came from. ( i noticed a bug: if a tap the "i" button and close the drawer, the attribution still shows on the map)
- URLs from tags in FeaturePanel (probably unrelated to this PR)
  - when opening URLs, would it not be better to open them in new windows instead ?

Dlurak commented:
The smaller and gray attribution doesn't really cause any optical clutter
And the urls aren't related to this issue.

![image](https://github.com/user-attachments/assets/89563dfb-001a-4d83-9d7e-4b83a85d2b36)

---

- I agree that the label "GeoURI" isn't ideal, but longer labels like "Open In App" aren't feasible due to the limited spaced. Do you maybe have a short label idea?
- The icon you proposed looks nice we can try it out :)


I did some quick experimentation for smaller icons and here is the result:

![image](https://github.com/user-attachments/assets/8b8b74f3-7838-47ff-a69c-99d2aaedb38f)

Smaller icons allow more options to fit without scrolling, which is a positive change. However, I believe the larger icons looked better visually and added clarity by allowing labels and icons that indicate link vs. share actions. Without these, users may struggle to differentiate services, particularly with the less intuitive GeoURI icon.

Dlurak commented:
![image](https://github.com/user-attachments/assets/ebd848d1-9537-4586-8c2a-38e20d61be0f)
@govvin what do you think of this new icon?

jvaclavik commented:
Hi @Dlurak, 
We were discussing dialog ux with @zbycz and we created following wireframe:

![image](https://github.com/user-attachments/assets/51e57cf5-0aeb-4bf2-944c-54c51c7bf99a)

We were thinking that current design is a too complex for the user and we could simplify it a bit. 
User has usually 3 use cases when he clicks to this button:
1. Copy current URL to clipboard or share it on mobile with sharing dialog
2. Get coordinates (in various formats)
3. Open in different app

Few thoughts:
- We agreed that it's not necessary to provide sharing dialog for all external apps. 
- link is in read only MUI Textfield component and you can copy or manually or with button. Also on mobile there is a share button for native dialog
- coordinates are similar like link but it's a MUI Select component where you can choose from multiple GPS formats (decimals vs degrees, etc.). You can copy or share it same way as in link section. 
- open in ... shows items in MUI List. First 3-4 will have icons, other one's shouldn't have it. When you click on it it opens it in external window/app but you can't share it in the native share dialog (design would be too cluttered). Also geouri should be another item in the list here. 

What do you think about it?

Dlurak commented:
![image](https://github.com/user-attachments/assets/7d0e1299-a74f-4d72-a556-e7956a3caf22)
Something like this?

Dlurak commented:
I think the new repo location broke preview deployments.

---

Overall I am happy with the code and the design as well, if you @govvin and @jvaclavik don't have any suggestions anymore I think we can merge it

jvaclavik commented:
> ![image](https://github.com/user-attachments/assets/7d0e1299-a74f-4d72-a556-e7956a3caf22)
> 
> Something like this?

Love it, ship it 👌

Btw input field in link could be worth the same size as select if possible. 

Could you please try to run again the deployment? For example just amend commit and force push. It should work again

Dlurak commented:
Both parts are now equally high and preview deployments work again, I'm gonna merge this :tada:

govvin commented:
> ![image](https://private-user-images.githubusercontent.com/84224239/382471018-2ea9015d-10de-475c-bb3c-a641db8eb972.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1OTA1ODksIm5iZiI6MTczMDU5MDI4OSwicGF0aCI6Ii84NDIyNDIzOS8zODI0NzEwMTgtMmVhOTAxNWQtMTBkZS00NzVjLWJiM2MtYTY0MWRiOGViOTcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAyVDIzMzEyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2MDQ2NTg2NTk0ZmRlMDExYTgwOGJhZTcwODg4MGQzNmYwNmJmZTJmOTFjNmE3M2I2ZjI3YWEwYmQ3MDk0MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ebCtgdK7CXG3nMFWN1Q36Df1IHj3Jql61y33tItJ2Q8) @govvin what do you think of this new icon and label?

@Dlurak , I wonder if you can use the icon and label for GeoURI in the revised feature panel, since you're using logos anyway? "Map App" is definitely more intuitive. :+1: 

In my opinion, it should be the first option since it ought to work natively in their device, followed by OSM, Mapy.cz and then Google. :smile: 

The new feature panel looks great, and straight-forward.
